### DISCUSSION TITLE
how to get Alighnment value for all buildings

### DISCUSSION THREAD

**Discussion by cdkang** [pysal/momepy#411] [Q&A] [answered]
Hello, 

I have tried to get Alignment value of 621,420 buildings as below. But, I have got the Alignment value of only 2,727 buildings. 
I think that the sw affected the results. Do you know how to get Alighnment value for all buildings? 

Sincerely,  

```py
queen = libpysal.weights.Queen.from_dataframe(buildings,silence_warnings=True,ids='uID')
sw = mm.sw_high(k=3, gdf=buildings, ids='uID', contiguity='queen')
buildings['alignment'] = mm.Alignment(buildings, sw, 'uID', 'orientation').series
```

**Comment by martinfleis** [comment, marked as answer]
Hello! Since you have created `Queen` weights based on buildings, the resulting weights matrix will be composed mostly of islands. With Queen contiguity, two buildings are considered neighbours if they share at least one vertex (point). You will need to find another way of defining spatial weights matrix to get a proper result of `Alignment`. 

One option is to create tessellation (see the docs) and use the Queen weights matrix derived from that but it may be tricky for such a large number of buildings. Other option is to create KNN or DistanceBand weights instead of Queen. Then you don't need to do the higher order.

```py
W = libpysal.weights.KNN.from_dataframe(df=buildings, ids=buildings['uID'], k=10)
 buildings['alignment'] = mm.Alignment(buildings, W, 'uID', 'orientation').series
```
You can obviously change the K to your liking.

**Comment by cdkang** [comment]
Your information is working well.

Thank you so much. 

Sincerely,

**Reply by jGaboardi** [reply]
@cdkang If @martinfleis's response answered your question please click "Mark as answer" there.
